src/log.cpp

```diff
@@ -390,7 +392,7 @@
 	if (m_cerr_buf_backup != nullptr)
 	{
 		std::cerr.rdbuf(m_cerr_buf_backup);
-		pthread_mutex_unlock(&cerr_log_redirection_mutex);
+		cerr_log_redirection_mutex.unlock();
```

Comment:
Regarding this and previous comment: the mutex needs to be locked in "redirect" method and released in "restore" method, because it needs to protect redirecting cerr conflicts between threads. Thread safety needs to be addressed in code that uses cerr_log_redirection. Do you suggest some improvements? Please analyze the issue and give some advice.

Reply:
You proposed to make locking inside "redirect" and "restore" independent. The original solution is locking mutex inside "redirect" and keep it locked until "restore" is called. You changed the functionality. The mutex is used to keep all code executed between "redirect" and "restore" single-threaded. In other scenario other thread can redirect already redirected cerr and make both logs incorrect. If you have idea how to keep similar feature but in more exception-safe way feel free to suggest.

Reply:
Please explain what will happen in case of exception, as std::unique_lock is in global scope.

Reply:
Good idea. However cerr_log_redirection destructor is calling "restore" and mutex is unlocked inside restore so I don't know if it will change anything. More complicated situation is when exception occurs inside "redirect" after mutex is locked or inside "restore" before mutex is unlocked.

Reply:
Implemented in [daa1420](https://github.com/docwire/docwire/pull/99/commits/daa1420e6c03916c5600160a579c760d950a99f4)